STUDIO/org.openl.security.acl/src/org/openl/security/acl/utils/AclPathUtils.java

```diff
@@ -0,0 +1,110 @@
+package org.openl.security.acl.utils;
+
+import java.util.List;
+
+import org.openl.rules.project.abstraction.AProjectArtefact;
+import org.openl.rules.project.impl.local.LocalRepository;
+import org.openl.rules.project.impl.local.ProjectState;
+import org.openl.rules.repository.api.FileData;
+import org.openl.rules.workspace.dtr.impl.FileMappingData;
+import org.openl.util.StringUtils;
+
+public final class AclPathUtils {
+
+    private AclPathUtils() {
+    }
+
+    /**
+     * Concatenates two paths.
+     *
+     * @param path1 first path
+     * @param path2 second path
+     * @return concatenated path
+     */
+    public static String concatPaths(String path1, String path2) {
+        if (StringUtils.isBlank(path1)) {
+            return path2;
+        }
+        if (StringUtils.isBlank(path2)) {
+            return path1;
+        }
+        var result = path1;
+        if (!result.endsWith("/")) {
+            result += "/";
+        }
```

Comment:
All folder pathes in OpenL must be ended by '/' by convention. It is better to put assertion of the input arguments there than do ad-hoc fix of them.


See https://github.com/openl-tablets/openl-tablets/blob/b7f6d0bf780d30efaca1928ed83df5c958f22289/STUDIO/org.openl.rules.repository/src/org/openl/rules/repository/api/Repository.java#L15-L22